Fruitties/shared/src/nativeMain/kotlin/com/example/fruitties/di/viewmodel/IOSViewModelOwner.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.fruitties.di.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelStore
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.lifecycle.viewmodel.CreationExtras
+import androidx.lifecycle.viewmodel.MutableCreationExtras
+import com.example.fruitties.di.AppContainer
+import com.example.fruitties.viewmodel.MainViewModel
+import kotlin.reflect.KClass
+
+class IOSViewModelOwner(private val appContainer: AppContainer) : ViewModelStoreOwner {
```

Comment:
Btw - does `ObservableObject` have it's own lifecycle that we need to account for? Can we just hook into init / deinit or is there something more specific based on what views are observing it? Is it possible that attaching ViewModel to AppContainer can cause it to leak?

Reply:
I don't think we need to deinit for our Observable classes (like UIModel). Here is the order / scope of where references are retained.

AppContainer (scope of the App).
IOSViewModelOwner (scope of the Window).
MainViewModel (scope of the View).
UIModel (scope of the View).

I don't see any obvious leaks, but let me know if you see something!

Reply:
I made some changes to remove the UIModel. I am using a SKIE preview API to bring StateFlow from KMP ViewModel to a SwiftUI State object. Since the app never deallocates anything on iOS, I think we're ok keeping everything as-is.